src/WormholeEndpoint.sol

```diff
@@ -31,6 +18,43 @@ abstract contract WormholeEndpoint is Endpoint {
     error InvalidWormholeSiblingZeroAddress();
     error InvalidWormholeSiblingChainIdZero();
 
+    /// =============== STORAGE ===============================================
+
+    bytes32 public constant WORMHOLE_CONSUMED_VAAS_SLOT =
+        bytes32(uint256(keccak256("whEndpoint.consumedVAAs")) - 1);
+
+    bytes32 public constant WORMHOLE_SIBLINGS_SLOT =
+        bytes32(uint256(keccak256("whEndpoint.siblings")) - 1);
+
+    function _getWormholeConsumedVAAsStorage()
+        internal
+        pure
+        returns (mapping(bytes32 => bool) storage $)
+    {
+        uint256 slot = uint256(WORMHOLE_CONSUMED_VAAS_SLOT);
+        assembly ("memory-safe") {
+            $.slot := slot
+        }
+    }
+
+    function _getWormholeSiblingsStorage()
+        internal
+        pure
+        returns (mapping(uint16 => bytes32) storage $)
+    {
+        uint256 slot = uint256(WORMHOLE_SIBLINGS_SLOT);
+        assembly ("memory-safe") {
+            $.slot := slot
+        }
+    }
+
+    // TODO -- fix this after some testing
+    uint256 constant _GAS_LIMIT = 500000;
+
+    address immutable _wormholeCoreBridge;
```

Comment:
should we keep immutables at the top of the contract?

Reply:
We can, I defaulted to keeping them above the constructor where they were initialized but can move all of them to the top of the contract.

Might actually make most sense to keep them right above the `_checkImmutables()` function.

Reply:
I actually moved them all to the top of the contract for consistency, since not all of the contracts with immutables have the `_checkImmutables()` function (e.g. `Manager.sol`)